### PR TITLE
Counter function

### PR DESCRIPTION
**Pull request by GabriellAnnunziata**

### CODE REVIEW

**Review by slavny11 — APPROVED** [review]

File: RayD/VitDTimeTracker.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
+//
+//  VitDTimeTracker.swift
+//  RayD
+//
+//  Created by Gabriella Annunziata on 23/05/23.
+//
+
+import Foundation
+
+class VitDTimeTracker: ObservableObject {
```
Comment (yukim17) [inline RayD/VitDTimeTracker.swift:10]: Good practice to put `final` keyword before `
class VitDTimeTracker..` if you are sure that you are not going to use inheritance for this class.

File: RayD/VitDTimeTracker.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    let vitDSuggested: Double = 10000.0
+
+    static func timeOfExposure (uVIndex: Int, exposedSkin: Int, skinType: Int, todayVitD: Double, VitDSuggested: Double) -> Int {
+        let vitDValue = Int(VitDSuggested - todayVitD)
+        let constant = 40
+        let time = (vitDValue * constant * skinType)/(uVIndex * exposedSkin * 21120)
+        return time
+    } // this is the initial value for the timer when starting a new session
+
+    func vitDCalculator (prodPerSeconds: Double, time: Int, todayVitD: Double, vitDSuggested: Double) {
```
Comment (palant-dev) [inline RayD/VitDTimeTracker.swift:25]: maybe we could remove prodPerSeconds and vitDSuggested if we are using the stored property inside the class accessing it with "self". Read the following comments for better clarity.

File: RayD/VitDTimeTracker.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        self.prodPerSeconds = (vitDSuggested * todayVitD)/Double(time)
+        }
+
+    func checkReachedGoal (todayVitD: Double, vitDSuggested: Double) -> Bool {
```
Comment (palant-dev) [inline RayD/VitDTimeTracker.swift:29]: Also in this function, we could remove vitDSuggested from the parameters of the function and use self inside the function body

File: RayD/VitDTimeTracker.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        var check = false
+        if todayVitD == vitDSuggested {
```
Comment (palant-dev) [inline RayD/VitDTimeTracker.swift:31]: Here we could use self.vitDSuggested

**Review by palant-dev — CHANGES_REQUESTED** [review]
Check the comments, maybe we could improve even more the functions. But great job, a lot of stuff!

**Review by palant-dev — CHANGES_REQUESTED** [review]
Check the comments, maybe we could improve even more the functions. But great job, a lot of stuff!

**Review by palant-dev — APPROVED** [review]
Everything is fine. Good job!

**Review by palant-dev — APPROVED** [review]
Everything is fine. Good job!

**Pull request merged**